File: code/mathematical-algorithms/gcd_and_lcm/gcd_and_lcm.kt
fun gcd(a: Int, b: Int): Int = if (b == 0) a else gcd(b, a % b)

fun lcm(a: Int, b: Int): Int = if (b == 0) 0 else a * b / gcd(a, b)

fun main(args: Array<String>) {
    val (a, b) = readLine()!!.split(' ').map(String::toInt)

    println("Enter two numbers separated by a space")
    println("GCD(" + a + ", " + b + "): " + gcd(a, b))
    println("LCM(" + a + ", " + b + "): " + lcm(a, b))
}

File: code/unclassified/GCDandLCM/GCDandLCM.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    cout<<"...GCD and LCM of numbers..."<<endl;
    int a,b;
    cout<<"Enter two numbers: "<<endl;
    cin>>a>>b;
    int gcd,lcm;
    gcd=__gcd(a,b);
    lcm=(a*b)/gcd;
    cout<<"The GCD of the two numbers is: "<<gcd<<endl;
    cout<<"The LCM of the two numbers is: "<<lcm;
    return 0;
}
